drivers/hwinfo/hwinfo_nrf.c

```diff
@@ -58,22 +58,56 @@ int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
 	if (reason & NRFX_RESET_REASON_DIF_MASK) {
 		flags |= RESET_DEBUG;
 	}
+	if (reason & NRFX_RESET_REASON_SREQ_MASK) {
+		flags |= RESET_SOFTWARE;
+	}
 
-#if !NRF_POWER_HAS_RESETREAS
+#if NRFX_RESET_REASON_HAS_CTRLAP
 	if (reason & NRFX_RESET_REASON_CTRLAP_MASK) {
 		flags |= RESET_DEBUG;
 	}
-	if (reason & NRFX_RESET_REASON_DOG0_MASK) {
-		flags |= RESET_WATCHDOG;
```

Comment:
Where has this gone? Isn't this present in nrf53?

Reply:
DOG0 is covered by `NRFX_RESET_REASON_DOG_MASK`.
See https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/helpers/nrfx_reset_reason.h#L140